model_api/cpp/models/src/classification_model.cpp

```diff
@@ -480,18 +500,15 @@ void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model
         if (outputShape.size() == 4 && (outputShape[ov::layout::height_idx(outputLayout)] != 1 ||
                                         outputShape[ov::layout::width_idx(outputLayout)] != 1)) {
             throw std::logic_error("Classification model wrapper supports topologies only"
-                                " with 4-dimensional output which has last two dimensions of size 1");
+                                " with 4-dimensional output hich has last two dimensions of size 1");
         }
 
         size_t classesNum = outputShape[ov::layout::channels_idx(outputLayout)];
         if (topk > classesNum) {
             throw std::logic_error("The model provides " + std::to_string(classesNum) + " classes, but " +
                                 std::to_string(topk) + " labels are requested to be predicted");
         }
-        if (classesNum == labels.size() + 1) {
-            labels.insert(labels.begin(), "other");
-            slog::warn << "Inserted 'other' label as first." << slog::endl;
-        } else if (classesNum != labels.size()) {
```

Comment:
This was the only place adding `other`. I think you've removed `other` for all models, not just for xai,

Reply:
That's right, I had to eliminate this ad-hoc heuristic completely.
Inserting `other` to the labels list requires some prior knowledge of the actual insertion place, which I'm not aware of, at least for OTX/Geti usecases.